### PR TITLE
feat(api/appeals): scaffolding of integration flows

### PR DESCRIPTION
**Pull request by fabfab-kc**
Initial scaffolding for main integrations

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/api/src/server/endpoints/integrations/integrations.controller.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { mapAppealFromTopic, mapDocumentFromTopic } from './integrations.mapper.js';
+
+/** @typedef {import('@pins/appeals/index.js').BlobInfo} BlobInfo */
+/** @typedef {import('@pins/appeals/index.js').AddDocumentsResponse} AddDocumentsResponse */
+/** @typedef {import('@pins/appeals.api').Schema.Folder} Folder */
+/** @typedef {import('express').RequestHandler} RequestHandler */
+
+/**
+ * @type {RequestHandler}
+ * @returns {Promise<any>}
```
Comment (shepda) [inline appeals/api/src/server/endpoints/integrations/integrations.controller.js:11]: This method could have a return type of `Promise<Response>` and then the `'@ts-ignore` wouldn't be needed below.

`Response` would need have a typedef like `/** @typedef {import('express').Response} Response */`

I've literally just changed several of these in the codebase :D

File: appeals/api/src/server/endpoints/integrations/integrations.controller.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	const data = mapAppealFromTopic(req.body);
+	const result = await createAppeal(data);
+
+	// @ts-ignore
+	return res.send(result);
+};
+
+/**
+ * @type {RequestHandler}
+ * @returns {Promise<any>}
```
Comment (shepda) [inline appeals/api/src/server/endpoints/integrations/integrations.controller.js:23]: This method could have a return type of `Promise<Response>` and then the `'@ts-ignore` wouldn't be needed below.

`Response` would need have a typedef like `/** @typedef {import('express').Response} Response */`

File: appeals/api/src/server/repositories/integrations.repository.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		// Save documents
+		// ...
+
+		return appeal;
+	});
+
+	return transaction;
+};
+
+export const createDocument = async (data) => {
```
Comment (shepda) [inline appeals/api/src/server/repositories/integrations.repository.js:35]: It doesn't look like this method needs `aync` as it's not doing anything asynchronous

Comment (fabfab-kc) [reply appeals/api/src/server/repositories/integrations.repository.js:35]: it's not doing anything at all at the moment :-)

Comment (shepda) [reply appeals/api/src/server/repositories/integrations.repository.js:35]: Exactly my point :D as that is currently how it behaves...however if necessary we can leave it on the expectation of future functionality

**Review by shepda — DISMISSED** [review]
LGTM

**Review by shepda — APPROVED** [review]

**Pull request merged**